pmd-modelica/etc/grammar/Modelica.jjt

```diff
@@ -0,0 +1,952 @@
+      )
+    )
+}
+
+void ImportClause(): {}
+{
+    <IMPORT>
+    (
+          (<IDENT> <EQ_SIGN> Name())
+        | (Name() <DOT> <ASTERISK>)
+        | (Name() <DOT> <LCURL> ImportList() <RCURL>)
+        | Name()
+    )
+    Comment()
+}
+
+void ImportList(): {}
+{
+    <IDENT>
+    (
+        <COMMA>
+        <IDENT>
+    )*
+}
+
+void ExtendsClause(): {}
+{
+    <EXTENDS>
+    Name()
+    (ClassModification())?
+    (Annotation())?
+}
+
+void ConstrainingClause(): {}
+{
+    <CONSTRAINEDBY>
+    Name()
+    (ClassModification())?
+}
+
+void ComponentClause(): {}
+{
+    TypePrefix()
+    TypeSpecifier()
+    (ArraySubscripts())?
+    ComponentList()
+}
+
+void TypePrefix(): {}
+{
+    (<FLOW> | <STREAM>)?
+    (<DISCRETE> | <PARAMETER> | <CONSTANT>)?
+    (<INPUT> | <OUTPUT>)?
+}
+
+void TypeSpecifier(): {}
+{
+    Name()
+}
+
+void ComponentList(): {}
+{
+    ComponentDeclaration()
+    (
+        <COMMA>
+        ComponentDeclaration()
+    )*
+}
+
+void ComponentDeclaration(): {}
+{
+    Declaration()
+    (ConditionAttribute())?
+    Comment()
+}
+
+void ConditionAttribute(): {}
+{
+    <IF>
+    Expression()
+}
+
+void Declaration(): {}
+{
+    <IDENT>
+    (ArraySubscripts())?
+    (Modification())?
+}
+
+void Modification(): {}
+{
+    (
+        ClassModification()
+        (
+            <EQ_SIGN>
+            Expression()
+        )?
+    ) | (
+        <EQ_SIGN>
+        Expression()
+    ) | (
+        <ASSIGN>
+        Expression()
+    )
+}
+
+void ClassModification(): {}
+{
+    <LPAREN>
+    (ArgumentList())?
+    <RPAREN>
+}
+
+void ArgumentList(): {}
+{
+    Argument()
+    (
+        <COMMA>
+        Argument()
+    )*
+}
+
+void Argument(): {}
+{
+      ElementModificationOrReplaceable()
+    | ElementRedeclaration()
+}
+
+void ElementModificationOrReplaceable(): {}
+{
+    (<EACH>)?
+    (<FINAL>)?
+    (
+          ElementModification()
+        | ElementReplaceable()
+    )
+}
+
+void ElementModification(): {}
+{
+    Name()
+    (Modification())?
+    StringComment()
+}
+
+void ElementRedeclaration(): {}
+{
+    <REDECLARE>
+    (<EACH>)?
+    (<FINAL>)?
+    (
+          (ShortClassDefinition() | ComponentClause1())
+        | ElementReplaceable()
+    )
+}
+
+void ElementReplaceable(): {}
+{
+    <REPLACEABLE>
+    (ShortClassDefinition() | ComponentClause1())
+    (ConstrainingClause())?
+}
+
+void ComponentClause1(): {}
+{
+    TypePrefix()
+    TypeSpecifier()
+    ComponentDeclaration1()
+}
+
+void ComponentDeclaration1(): {}
+{
+    Declaration()
+    Comment()
+}
+
+void ShortClassDefinition(): {}
+{
+    ClassPrefixes()
+    ShortClassSpecifier()
+}
+
+void EquationSection(): {}
+{
+    (<INITIAL>)?
+    <EQUATION>
+    (
+        Equation()
+        <SC>
+    )*
+}
+
+void AlgorithmSection(): {}
+{
+    (<INITIAL>)?
+    <ALGORITHM>
+    (
+        Statement()
+        <SC>
+    )*
+}
+
+void Equation(): {}
+{
+    (
+          (SimpleExpression() <EQ_SIGN> Expression())
```

Comment:
This looks like it needs its own node, eq `RegularEquation` or whatever that is.